rfcs/min-relations.md

```diff
@@ -0,0 +1,644 @@
+
+```rust
+fn frenemies(
+    query: Query<Entity, (With<Relation<FriendsWith>>, With<Relation<EnemiesWith>>)>,
+    player: Res<Player>,
+) {
+    query
+        .filter_relation(RelationFilter::<FriendsWith>::target(player.entity))
+        .filter_relation(RelationFilter::<EnemiesWith>::target(player.entity))
+        .for_each(|entity| println!("{} is frenemies with the player!", entity));
+}
+```
+
+Filters chained in this way will operate on the restricted list produces by the previous filter (following "and semantics").
+We can even combine positive and negative filters by including multiple copies of the same relation in our query, and then specifying which relation we're filtering using the second type parameter.
+This advanced technique is helpful when you really need to be specific, as shown in these examples:
+
+```rust
+fn herbivores(
+    mut commands: Commands,
+    plants_query: Query<Entity, With<Plant>>,
+    animals_query: Query<Enity, With<Animal>>,
+    species_query: Query<Entity, (With<Relation<Consumes>>, Without<Relation<Consumes>>)>,
+) {
+    let plants = plants_query.iter().collect();
+    let animals = animals_query.iter().collect();
+
+    // The generics on `filter_relation` are normally inferred but we can manually
+    // specify them to disambiguate which `Relation<T>` we are filtering in the query
+    species_query
+        // The second type parameter controls which relation filter of that type is being modified
+        .filter_relation::<Consumes, InFilter<{ 0 }>>(RelationFilter::any_of().targets(plants))
+        // Because this is a without filter, this means that we exclude all entities
+        // with even one Consumes relation that targets an animal
+        .filter_relation::<Consumes, InFilter<{ 1 }>>(RelationFilter::any_of().targets(animals))
+        .for_each(|entity| commands.entity(entity).insert(Herbivore));
+}
+
+```
+
+### Grouping entities
+
+By targeting a common entity, relations work fantastically as an ergonomic way to group entities.
+This pattern acts as a nice complement to marker components when either:
+
+1. You want to define the *kind* of group, rather than just membership.
+2. The number of different groups you need is unknowable at compile time.
+
+Here are some concrete examples where this pattern works well:
+
+- you're storing multiple parts (such as meshes or sprites with their own `Transform`s) of the same in-game-object across multiple entities
+- you want to dynamically group units together, like in a real-time-strategy game
+- the effects created by a single entity, like the bullets they're spewing out
+- you want a way to mark the controller of a unit in a queryable way, but the number of possible controllers is unknown at compile time
+- you have multiple cameras, and want to mark which frustum each entity is in so they can be culled independently
+
```

Comment:
Per feedback, this needs a simple worked example to show users what this looks like in practice :p